ddtrace/tracer/tracer.go

```diff
@@ -163,9 +163,14 @@ func SetUser(s Span, id string, opts ...UserMonitoringOption) {
 	}
 	if span, ok := s.(*span); ok && span.context != nil {
 		span = span.context.trace.root
+		// Unset the propagated user ID by default so that if the function is called without WithPropagation(),
+		// a propagated user ID coming from upstream won't be propagated anymore.
+		// RFC: https://docs.google.com/document/d/1T3qAE5nol18psOaHESQ3r-WRiZWss9nyGmroShug8ao/edit#heading=h.3wmduzc8mwe1
+		delete(span.context.trace.propagatingTags, keyPropagatedUserID)
```

Comment:
Now that this code is not just using the thread-safe `SetTag` I believe it needs to lock the span that is being modified.

Reply:
42f2d4b9e0